common/djangoapps/enrollment/views.py

```diff
@@ -469,6 +472,14 @@ def get(self, request):
                     ).format(username=username)
                 }
             )
+        if username == request.user.username or self.has_api_key_permissions(request):
+            return Response(enrollment_data)
+        filtered_data = []
+        for enrollment in enrollment_data:
```

Comment:
Couple of changes that can be made to reduce SQL queries:

* check whether the user is global_staff on L475, in this case the data won't need to be filtered and you can return the whole result.
* before looping through `enrollment_data`, fetch a list of all the courses for which the current user is staff (in a single db operation).  then during loop iterations, just check if the course key is in that list, rather than consulting `user_has_role()`.

I'm not aware of a function existing presently that can list all the course/keys for which a given user is staff, but I've heard rumors that @dsjen is working on something similar or related.   If nothing else you might end up creating such a function in [auth.py](https://github.com/open-craft/edx-platform/blob/master/common/djangoapps/student/auth.py).